R/dgp-lib-y.R

```diff
@@ -31,23 +29,43 @@
 #' y <- generate_y_linear(X = X, U = U, betas = c(3, -1), betas_unobs = c(1, 2))
 #' 
 #' @export
-generate_y_linear <- function(X, U, betas = 0, betas_unobs = 0, intercept = 0,
-                              err = NULL, return_support = FALSE, ...) {
+generate_y_linear <- function(X, U, betas = NULL, betas_unobs = NULL,
+                              intercept = 0, err = NULL, return_support = FALSE,
+                              ...) {
   n <- nrow(X)
   p <- ncol(X)
-  
+
   if (missing(U)) {
     U <- matrix(0, nrow = n, ncol = 1)
   }
-  
-  betas <- generate_coef(betas = betas, p = p)
-  betas_unobs <- generate_coef(betas = betas_unobs, p = ncol(U),
-                               betas_name = "betas_unobs")
-  
-  eps <- generate_errors(err = err, n = n, X = X, ...)
-  
-  y <- intercept + c(as.matrix(U) %*% betas_unobs + as.matrix(X) %*% betas + eps)
-  
+
+  fun_args <- dots_to_fun_args(prefix = c("err", "betas", "betas_unobs"), ...)
+  err_args_list <- fun_args$.err_args
+  betas_args_list <- fun_args$.betas_args
+  betas_unobs_args_list <- fun_args$.betas_unobs_args
+  optional_args_list <- fun_args$.optional_args
+
+  betas <- R.utils::doCall(
+    generate_coef, .betas = betas, .p = p,
+    args = c(betas_args_list, optional_args_list), .ignoreUnusedArgs = FALSE
```

Comment:
Ya, I don't think it will be necessary to use `.betas_.s` unless the user for some reason also wants to pass an `.s` parameter to `err`, which does seem rare. But in that case, using `.betas_.s` and `.err_.s` should work (though awkward).

The original reason for add `.` to `generate_coef()`'s args was because of conflicts between its `sd` argument and if the user also passed e.g. `rnorm()` as `err` to `linear_gaussian_dgp()`, where we had `betas_sd`. So from there user of `linear_gaussian_dgp()` there shouldn't be a conflict by passing `betas_sd` and `sd`, but without some complicated filtering on `optional_args_list` both `sd`s would get passed to `generate_coef()`. 

Removing the `sd` arg from `generate_coef()` may have reduced the need for `.`, but I wonder if added `.` to the args of all of the "building block" functions (e.g. `generate_X_gaussian()`) might help to avoid unintended consequences (like passing `sd` for `err`, but it also would get picked up by `generate_X_gaussian()`). Let me know what you think.

Reply:
I like the way you're using `.` right now, so we should keep it as is. I'm ok with the rare cases where the syntax is awkward since it still works as expected.